erpnext/stock/doctype/purchase_receipt/purchase_receipt.py

```diff
@@ -530,3 +531,27 @@ def make_purchase_return(source_name, target_doc=None):
 def update_purchase_receipt_status(docname, status):
 	pr = frappe.get_doc("Purchase Receipt", docname)
 	pr.update_status(status)
+
+@frappe.whitelist()
+def make_stock_entry(source_name,target_doc=None):
+	def set_missing_values(source, target):
+		if source.name:
+			target.stock_entry_type = "Material Transfer"
+		for item in target.items:
+			for pr_item in source.items:
+				if pr_item.item_code == item.item_code:
+					item.s_warehouse = pr_item.warehouse
+			item.reference_purchase_receipt = source.name
```

Comment:
Instead of line 540 - 544, you can use below code in the 552-554

"Purchase Receipt Item": {
	"doctype": "Stock Entry Detail",
	"field_map": {
		"warehouse": "s_warehouse",
		"name": "reference_purchase_receipt"				
	},
},

Reply:
done